server/server.go

```diff
@@ -452,6 +446,56 @@ func (h RequestHandler) NewThreadRoute(res http.ResponseWriter, req *http.Reques
 	}
 }
 
+func (h RequestHandler) DeletePostRoute(res http.ResponseWriter, req *http.Request) {
+	if req.Method == "GET" {
+		IndexRedirect(res, req)
+		return
+	}
+	threadURL := req.PostFormValue("thread")
+	postid, ok := util.GetURLPortion(req, 3)
+	loggedIn, userid := h.IsLoggedIn(req)
+
+	if !loggedIn || !ok {
+		title := "Unaccepted request"
+		data := GenericMessageData{
+			Title:       title,
+			Message:     "The post you tried to delete was not found, or you were not allowed to delete it",
+			LinkMessage: "Go back to",
+			Link:        threadURL,
+			LinkText:    "the thread",
+		}
+		h.renderView(res, "generic-message", TemplateData{Data: data, LoggedIn: loggedIn})
+		return
+	}
+
+	post, err := h.db.GetPost(postid)
+	authorized := post.AuthorID == userid
+	if !authorized || err != nil {
+		title := "Unaccepted request"
+		data := GenericMessageData{
+			Title:       title,
+			Message:     "You were not allowed to delete the post",
```

Comment:
This message obfucates the `GetPost` err - it might be nice to render two messages depending on a get post failure or a actual `!authorized`

Also maybe you should shouldn't check`post.AuthorID` until you know err is nil ? Otherwise could be nil access I think

Reply:
appreciate the comment!

> Also maybe you shouldn't checkpost.AuthorID until you know err is nil ? Otherwise could be nil access I think

good point about not checking it prematurely; GetPost will return an empty `type Post struct{...}` if error, so it wouldn't be a nil access i think, but i'll restructure anyway just to be on the safe side with AuthorIDs colliding with the default value on `Post` :)

Reply:
Ah I see. Makes sense!